index.js

```diff
@@ -474,4 +475,32 @@ var ab2hex = function (ab) {
   return new Buffer(buffer).toString('hex')
 }
 
+// wrap around webcrypto.getRandomValues() to use high-quality PRNG, when available
+var getRandomValues = function (ab) {
+  var err, i, j, octets
+
+  if (!ab.BYTES_PER_ELEMENT) {
+    err = new Error()
+    err.name = 'TypeMisMatchError'
+    throw err
+  }
+  if (ab.length > 65536) {
+    err = new Error()
+    err.name = 'QuotaExceededError'
+    throw err
+  }
+
+  octets = crypto.randomBytes(ab.length * ab.BYTES_PER_ELEMENT)
+
+  if (ab.BYTES_PER_ELEMENT === 1) ab.set(octets)
+  else {
```

Comment:
my concern wasn't about performance but rather about having to reason about correctness in core crypto code.

i think it's easy to prevent people getting burnt by putting `@param Uint8Array` in jsdoc for this function so it's obvious what the expected input is. you could even rename it to something like `getRandomValuesFromUint8Array`. either way, i don't think it's reasonable for people to see a function named `getRandomValues` and automatically assume it's a full polyfill for window.crypto.getRandomValues.